core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ByteArrayAsyncResponseTransformer.java

```diff
@@ -60,11 +63,30 @@ public void onStream(SdkPublisher<ByteBuffer> publisher) {
         publisher.subscribe(new BaosSubscriber(cf));
     }
 
+    public void onStream(SdkPublisher<ByteBuffer> publisher, long maximumBufferSize) {
```

Comment:
Hmm, not sure if we want to add another onStream here, why do we need this?

Reply:
Was thinking to use maximumBufferSize for large part sizes, but ByteArrayAsyncResponseTransformer loads everything into memory anyways so its redundant. Will revert this addition